### PR TITLE
feat: handle not/enough fee asset balance for network fees at final trade quote time

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Does what it says on the box - this ensures that users aren't able to execute (and either end up with estimate failures if using native, or being stuck at signing time if using e.g MM) trades without enough balance for gas.

Note, this isn't specific to Portals only: 
- as we refetch network fees at interval, this will apply to other swappers too
- for any swapper currently, this will also work with any swapper if you fund/drain native asset from addy during the swap

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/8268

## Risk

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

Low, ensure you don't see this when you shouldn't

## Testing

- For Portals swaps, not enough balance should be caught and the button should be disabled, in error state. Funding your account with enough native asset should make the button green (blue) again
- For other swappers, you can also test this by

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

## Screenshots (if applicable)

- Enough asset balance is happy

https://jam.dev/c/b5a9bc43-7e0d-468a-8c3a-35013c4e9d8a

- Not enough asset balance is happy

https://jam.dev/c/38a46118-9851-4a51-85f3-7302be4309ae

- Draining native asset makes button go sad

https://jam.dev/c/4733177e-d628-4c0b-8f17-32e370b40b87

- Funding native asset makes button go happy

https://jam.dev/c/6d841877-c23a-42e9-9faf-0c3eade9a999

### CODE REVIEW

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx [hunk 1]
```diff
@@ -155,20 +160,54 @@ export const HopTransactionStep = ({
...
+    // This should not happen at final quote time but we need to content TS
+    if (!networkFeeCryptoBaseUnit) return true
+
+    return bn(feeAssetBalance).gte(networkFeeCryptoBaseUnit)
+  }, [feeAssetBalance, tradeQuoteStep])
+
+  const signButtonCopy = useMemo(() => {
+    if (!hasEnoughBalanceForNetworkFees)
+      return translate('modals.send.errors.notEnoughNativeToken', {
+        asset: feeAsset!.symbol,
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx:191]: Actually defined if we end up here

**Review by woodenfurniture — APPROVED** [review]

**Review by woodenfurniture — CHANGES_REQUESTED** [review]
Requesting changes to revok accidental approval.

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx [hunk 2]
```diff
@@ -155,20 +160,54 @@ export const HopTransactionStep = ({
 
   const { isFetching, data: tradeQuoteQueryData } = useGetTradeQuotes()
 
+  const feeAsset = useAppSelector(state =>
+    selectFeeAssetByChainId(state, tradeQuoteStep?.sellAsset.chainId ?? ''),
+  )
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx:165]: Try it, you'll like it :P

```suggestion
  const feeAsset = useSelectorWithArgs(selectFeeAssetById, orderToCancel?.sellAssetId ?? '')
```

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx [hunk 3]
```diff
@@ -155,20 +160,54 @@ export const HopTransactionStep = ({
...
+  const feeAssetBalanceFilter = useMemo(
+    () => ({ assetId: feeAsset?.assetId ?? '', accountId: sellAssetAccountId ?? '' }),
+    [feeAsset?.assetId, sellAssetAccountId],
+  )
+  const feeAssetBalance = useAppSelector(state =>
+    selectPortfolioCryptoBalanceBaseUnitByFilter(state, feeAssetBalanceFilter),
+  )
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx:172]: May as well try to suggest it

```suggestion
  const feeAssetBalance = useSelectorWithArgs(selectPortfolioCryptoBalanceBaseUnitByFilter, feeAssetBalanceFilter)
```

**Review by woodenfurniture — APPROVED** [review]
First pas, looking good. Needs runtime testing (to follow)

**Review by woodenfurniture — CHANGES_REQUESTED** [review]
# Testing notes

❌ Does not appear to be taking the gas fee into account when doing a ETH -> FOX trade on ethereum:

https://jam.dev/c/c86fe6b6-ae5a-44fd-bb1e-79bd5e62688f

MM is catching this, which confirms the lack of balance for gas:

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx (hunk 3 above)
Comment (gomesalexandre) [reply src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx]: hell yeah 2751c96d3a

File: src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/HopTransactionStep.tsx]: hell yeah 2751c96d3a

**Comment by gomesalexandre** [comment]
> ❌ Does not appear to be taking the gas fee into account when doing a ETH -> FOX trade on ethereum:

Very nice catch ser, we indeed need to check for the Tx value as well in the case of native sells 🧠 286d25dacae8d1ad3d7b620649d73a756067fe32
https://jam.dev/c/79d024d3-8d12-45af-aa5a-6aafe1310f28

**Review by woodenfurniture — APPROVED** [review]
Retested, native balance checks are worky again 🎉

**Pull request merged**